keps/sig-node/127-user-namespaces/README.md

```diff
@@ -335,6 +335,8 @@ bool `pod.spec.hostUsers`.
 The mapping length will be 65536, mapping the range 0-65535 to the pod. This wide
 range makes sure most workloads will work fine. Additionally, we don't need to
 worry about fragmentation of IDs, as all pods will use the same length.
+The mapping length (multiple of 65536) will be customizable via a new
+`KubeletConfiguration` property `subidsPerPod`.
```

Comment:
I'd got the impression we might want to make the mapping size configurable on a per-Pod basis.

Reply:
(what if you have a particular Pod that assigns a (POSIX) ID to each user, and you have 42000000 users, but all your other Pods only need 65000 UIDs?)

Reply:
I think it's possible but not a common case IMO, and the implementation of adding a pod API field would be much more complex than adding a kubelet configuration field. I'm not sure the maintenance burden is worth it

Reply:
So long as we're not accidentally tying ourselves into _not_ being able to extend the Pod API in the future. If we are tying ourselves, let's make sure we'd never want the option.

Reply:
What about introducing a Pod security context property like `securityContext.userNS.staticMappingWithUsername: "foo"`.
This will run `getsubids foo` to obtain the subID range, and assign the entire range to the Pod.
(So, this is different from `getsubids kubelet` which returns the total range for the 110 pods)

Multiple pods may use the same range at their own risk.
This allows assigning an extremely large subID range. $(2^{32}-65536)$ at maximum.

Reply:
is the max idrange inside of a container flexible? as in: could we have a kubelet field that toggles a dynamic range and the runtime interpret the range in the image?

Reply:
how to prevent that such a field is not abused?  An image could claim all the available IDs and prevents that other pods can be created

Reply:
Admission-time checks is where I'd start; also ResourceQuota and LimitRange specifically.